qiskit/providers/backend.py

```diff
@@ -229,3 +241,445 @@ class can handle either situation.
+        name: str = None,
+        description: str = None,
+        online_date: datetime.datetime = None,
+        backend_version: str = None,
+        **fields,
+    ):
+        """Initialize a BackendV2 based backend
+
+        Args:
+            provider: An optional backwards reference to the
+                :class:`~qiskit.providers.Provider` object that the backend
+                is from
+            name: An optional name for the backend
+            description: An optional description of the backend
+            online_date: An optional datetime the backend was brought online
+            backend_version: An optional backend version string. This differs
+                from the :attr:`~qiskit.providers.BackendV2.version` attribute
+                as :attr:`~qiskit.providers.BackendV2.version` is for the
+                abstract :attr:`~qiskit.providers.Backend` abstract interface
+                version of the object while ``backend_version`` is for
+                versioning the backend itself.
+            fields: kwargs for the values to use to override the default
+                options.
+
+        Raises:
+            AttributeError: If a field is specified that's outside the backend's
+                options
+        """
+
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+        self._basis_gates_all = None
+        self.name = name
+        self.description = description
+        self.online_date = online_date
+        self.backend_version = backend_version
+
+    @property
+    def instructions(self) -> List[Tuple[Instruction, Tuple[int]]]:
+        """A list of Instruction tuples on the backend of the form ``(instruction, (qubits)``"""
+        return self.target.instructions
+
+    @property
+    def operations(self) -> List[Instruction]:
+        """A list of :class:`~qiskit.circuit.Instruction` instances that the backend supports."""
+        return list(self.target.operations)
+
+    def _compute_incomplete_basis(self):
+        incomplete_basis_gates = []
+        all_permutations_all_size = {}
+        for inst, qarg_props in self.target.items():
+            qargs = set(qarg_props)
+            size = len(next(iter(qargs)))
+            if size not in all_permutations_all_size:
+                all_permutations_all_size[size] = set(
+                    itertools.permutations(range(self.target.num_qubits), size)
+                )
+            if qargs != all_permutations_all_size[size]:
+                incomplete_basis_gates.append(inst)
+        self._basis_gates_all = incomplete_basis_gates
+
+    @property
+    def operation_names(self) -> List[str]:
```

Comment:
```suggestion
    def instruction_names(self) -> List[str]:
```
Mixing instruction and operation is a bit confusing to me, unless there is some more long term plan to call things `Operation` in circuit API?

Reply:
This is part of the plan to move the current `Instruction` class to be called `Operation` and the name `Instruction` to be what is represented by `(Instruction, qargs, cargs)` today. See: https://github.com/Qiskit/qiskit-terra/pull/5885#discussion_r747064724 where @nkanazawa1989 asked the same thing